### DISCUSSION TITLE
Autoincrementing column within another column

### DISCUSSION THREAD

**Discussion by chriskuech** [prisma/prisma#18521] [Q&A]
```prisma
model Widget {
  orgId String
  key Int @default(autoincrement())

  @@unique([orgId, key])
}
```

Is there a prismatic way to modify this model such that `key` autoincrements but only within a given `orgId`. In other words, for every `orgId`, `key` autoincrements starting at 1 (rather than a single global autoincrement).

**Comment by ludralph** [comment]
Hi @chriskuech 👋 

This is not currently possible using Prisma ORM. You need to use [raw query](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access) to achieve this. You can also open a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) and document your usecase so our engineerign team can take a look

**Reply by ludralph** [reply]
Hi @chriskuech 

Do you have any more questions around this issue? If no, it would be really helpful if you could quickly **mark this Discussion as _answered_** to indicate that your issue has been resolved. 

In case you do have any outstanding questions, feel free to let us know and we'd be happy to help!
